### PR TITLE
[Backport Foxy] Update QD

### PR DESCRIPTION
**Pull request by ahcorde**

### CODE REVIEW

**Review by brawner — APPROVED** [review]

File: rcl_logging_spdlog/QUALITY_DECLARATION.md [hunk 1]
```diff
@@ -109,28 +128,36 @@ There is an automated test which runs a linter that ensures each file has at lea
...
+The `rcutils` package provides an API which contains common utilities and data structures useful when programming in C.
+
+It is **Quality Level 2**, see its [Quality Declaration document](https://github.com/ros2/rcutils/blob/foxy/QUALITY_DECLARATION.md).
+
+#### `spdlog_vendor`
+
+The `spdlog_vendor` package provides a CMake shim over the spdlog library.
 
-`spdlog_vendor` was declared to be Quality Level 4 [here](https://github.com/ros2/spdlog_vendor/blob/master/Quality_Declaration.md).
+It is **Quality Level 3**, see its [Quality Declaration document](https://github.com/ros2/spdlog_vendor/blob/foxy/QUALITY_DECLARATION.md).
```
Comment (chapulina) [inline rcl_logging_spdlog/QUALITY_DECLARATION.md:152]: `spdlog_vendor` is now level 1, see https://github.com/ros2/spdlog_vendor/pull/17. We can update this here and bump `rcl_logging_spdlog` to level 2. It's now only pending on `rcutils` to go to level 1.

Not sure if the right way would be to do this on `master` and backport.

Comment (ahcorde) [reply rcl_logging_spdlog/QUALITY_DECLARATION.md]: I will forward port the commit https://github.com/ros2/rcl_logging/pull/61/commits/1360fdba9d2b8995ba58efeb8576700a057ca4ed

File: rcl_logging_spdlog/QUALITY_DECLARATION.md [hunk 2]
```diff
@@ -109,28 +128,36 @@ There is an automated test which runs a linter that ensures each file has at lea
...
+It is **Quality Level 1**, see its [Quality Declaration document](https://github.com/ros2/spdlog_vendor/blob/foxy/QUALITY_DECLARATION.md).
 
 ### Optional Direct Runtime ROS Dependencies [5.ii]
 
 `rcl_logging_spdlog` has no optional Direct Runtime ROS dependencies that need to be considered for this declaration.
 
 ### Direct Runtime non-ROS Dependency [5.iii]
 
-`rcl_logging_spdlog` has a Direct Runtime non-ROS dependenciy the `spdlog` library. It was declared to be Quality Level 4 [here](https://github.com/ros2/spdlog_vendor/blob/master/SPDLOG_QUALITY_DECLARATION.md).
+`rcl_logging_spdlog` has a Direct Runtime non-ROS dependenciy the `spdlog` library. It was declared to be Quality Level 4 [here](https://github.com/ros2/spdlog_vendor/blob/foxy/SPDLOG_QUALITY_DECLARATION.md).
```
Comment (chapulina) [inline rcl_logging_spdlog/QUALITY_DECLARATION.md:160]:
```suggestion
`rcl_logging_spdlog` has a Direct Runtime non-ROS dependenciy the `spdlog` library. It was declared to be Quality Level 1 [here](https://github.com/ros2/spdlog_vendor/blob/foxy/SPDLOG_QUALITY_DECLARATION.md).
```

File: rcl_logging_spdlog/QUALITY_DECLARATION.md [hunk 3]
```diff
@@ -2,7 +2,7 @@ This document is a declaration of software quality for the `rcl_logging_spdlog`
 
 # rcl_logging_spdlog Quality Declaration
 
-The package `rcl_logging_spdlog` claims to be in the **Quality Level 4** category.
+The package `rcl_logging_spdlog` claims to be in the **Quality Level 3** category.
```
Comment (chapulina) [inline rcl_logging_spdlog/QUALITY_DECLARATION.md:5]: It's level 2, pending `rcutils` to be level 1.

```suggestion
The package `rcl_logging_spdlog` claims to be in the **Quality Level 2** category.
```

**Review by chapulina — APPROVED** [review]

**Pull request merged**